src/components/OrderItem/OrderItem.jsx

```diff
@@ -1,30 +1,36 @@
 import React from 'react';
+import IconButton from '@material-ui/core/IconButton';
+import DeleteIcon from '@material-ui/icons/Delete';
 import { Price } from '..';
 
-export class OrderItem extends React.Component {
-  constructor(props) {
-    super(props);
-    this.action = this.action.bind(this);
-  }
-
-  action() {
-    const { id, removeFromOrder } = this.props;
+export const OrderItem = ({
+  id,
+  removeFromOrder,
+  name,
+  price,
+  bread,
+  isHot,
+  hasButton,
+  orderOwner
+}) => {
+  const handleClick = () => {
     removeFromOrder(id);
-  }
-
-  render() {
-    const { name, price, bread, isHot } = this.props;
+  };
 
-    return (
-      <li className="order-item">
-        <p className="order-item__name">
-          {name}
-          <span className="order-item__price"><Price value={price} /></span>
-        </p>
-        {isHot && <small className="hot">HOT</small>}
-        {bread !== 'Standard' && <small>{bread}</small>}
-        <button type="button" onClick={this.action}>- Remove</button>
-      </li>
-    );
-  }
-}
+  return (
+    <li className="order-item">
+      {orderOwner && <p className="order-item__owner">{orderOwner}</p>}
+      <p className="order-item__name">
+        {name}
+        <span className="order-item__price"><Price value={price} /></span>
+      </p>
+      {isHot && <small className="hot">HOT</small>}
+      {bread !== 'Standard' && <small>{bread}</small>}
+      {hasButton && (
+        <IconButton aria-label="Delete" onClick={handleClick}>
```

Comment:
You could inline `handleClick` as an anonymous function on `onClick` then you can implicitly return 👍 

`<IconButton aria-label="Delete" onClick={() => removeFromOrder(id)}>`